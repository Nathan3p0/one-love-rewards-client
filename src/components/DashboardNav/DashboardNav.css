.dashboard__menu {
    height: 100%;
}
.dashboard__menu ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    width: 100%;
    height: 100%;
    background-color: rgb(75, 89, 102);
}

.dashboard__menu li a {
    display: block;
    color: #fff;
    padding: 1rem 1rem;
    text-decoration: none;
    border-top: 1px solid rgb(106, 126, 146);
}

.dashboard__menu li a:hover {
    background-color: rgb(106, 126, 146);
}

.dashboard__menu .is-active {
    background-color: rgb(106, 126, 146);
}