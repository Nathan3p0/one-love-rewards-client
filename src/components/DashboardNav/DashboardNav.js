import React from 'react'
import { Link } from 'react-router-dom'

const DashboardNav = () => {
    return (
        <nav className="dashboard__menu">
            <Link to='/dashboard/redeem'>
                <div className="dashboard__menu-option redeem">
                    <p>Redeem Points</p>
                </div>
            </Link>
            <Link to='/dashboard/add-points'>
                <div className="dashboard__menu-option add-points">
                    <p>Add Points</p>
                </div>
            </Link>
            <Link to='/dashboard/edit-rewards'>
                <div className="dashboard__menu-option edit">
                    <p>Edit Rewards</p>
                </div>
            </Link>
            <Link to='/dashboard/add-member'>
                <div className="dashboard__menu-option add-member">
                    <p>Add Member</p>
                </div>
            </Link>
            <Link to='/dashboard/members'>
                <div className="dashboard__menu-option see-all">
                    <p>See All Members</p>
                </div>
            </Link>
            <Link to='/dashboard/delete-member'>
                <div className="dashboard__menu-option delete">
                    <p>Delete Member</p>
                </div>
            </Link>
        </nav>
    )
}

export default DashboardNav