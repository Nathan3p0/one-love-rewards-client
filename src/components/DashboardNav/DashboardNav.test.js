import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import DashboardNav from './DashboardNav'
import { BrowserRouter } from 'react-router-dom'

describe('Dashboard Nav Component Rendering Tests', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><DashboardNav /></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders to UI as expected', () => {
        const tree = renderer.create(<BrowserRouter><DashboardNav /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('renders to UI as expected with PROPS', () => {
        const tree = renderer.create(<BrowserRouter><DashboardNav
            handleLogout={() => { }}
            toggleNav={() => { }}
        /></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })

})