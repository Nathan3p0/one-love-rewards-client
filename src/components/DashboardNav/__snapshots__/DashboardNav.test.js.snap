// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Nav Component Rendering Tests renders to UI as expected 1`] = `
<nav
  className="dashboard__menu"
  role="navigation"
>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-points"
        onClick={[Function]}
      >
        Add & Redeem Points
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-member"
        onClick={[Function]}
      >
        Add Member
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/members"
        onClick={[Function]}
      >
        See All Members
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/delete-member"
        onClick={[Function]}
      >
        Delete Member
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Dashboard Nav Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<nav
  className="dashboard__menu"
  role="navigation"
>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-points"
        onClick={[Function]}
      >
        Add & Redeem Points
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-member"
        onClick={[Function]}
      >
        Add Member
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/members"
        onClick={[Function]}
      >
        See All Members
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/delete-member"
        onClick={[Function]}
      >
        Delete Member
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected 1`] = `
<nav
  className="dashboard__menu"
  role="navigation"
>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-points"
        onClick={[Function]}
      >
        Add & Redeem Points
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-member"
        onClick={[Function]}
      >
        Add Member
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/members"
        onClick={[Function]}
      >
        See All Members
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/delete-member"
        onClick={[Function]}
      >
        Delete Member
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Member List Item Component Rendering Tests renders to UI as expected with PROPS 1`] = `
<nav
  className="dashboard__menu"
  role="navigation"
>
  <ul>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-points"
        onClick={[Function]}
      >
        Add & Redeem Points
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/add-member"
        onClick={[Function]}
      >
        Add Member
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/members"
        onClick={[Function]}
      >
        See All Members
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/dashboard/delete-member"
        onClick={[Function]}
      >
        Delete Member
      </a>
    </li>
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
